### ISSUE TITLE
[Bug] Positions are not calculated correctly during the decoding phase

### ISSUE THREAD

**Issue by jimoosciuc** [sgl-project/sglang-jax#87]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.
- [ ] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

Positions are not calculated correctly during the decoding phase, affecting accuracy

### Reproduction

python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen-7B-Chat --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.4 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 64 --jax-precompile-prefill-token-paddings 4096 --max-running-requests 64 --skip-server-warmup --attention-backend=fa --max-prefill-tokens=4096

evalscope eval  --model Qwen-7B-Chat --api-url http://127.0.0.1:30000/v1 --api-key EMPTY --eval-type openai_api --datasets gsm8k --eval-batch-size 8

### Environment

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache
